### PR TITLE
apm-data: set codec: best_compression for logs-apm.* data streams

### PR DESCRIPTION
**Pull request by endorama**
Set `codec: best_compression` on logs related data streams for `apm-data` plugin. This aligns the configuration to what happens when using the `apm` integration package.

This setting is set by Fleet (see [issue](https://github.com/elastic/kibana/issues/160288)) to reduce storage for ingested data. We want to retain the same behaviour with the integration. 

Note that `best_compression` codec is already used for `metrics` but not for `traces`, for which we will evaluate enabling it at a later moment.

Rel: https://github.com/elastic/apm-server/issues/11529

### CODE REVIEW

File: x-pack/plugin/apm-data/src/main/resources/component-templates/logs-apm@settings.yaml
Comment (axw) [inline x-pack/plugin/apm-data/src/main/resources/component-templates/logs-apm@settings.yaml:1]: This new template needs to be added to `x-pack/plugin/apm-data/src/main/resources/resources.yaml`. The version in that file needs to be bumped too, so we update previously installed index templates.

**Review by axw — COMMENTED** [review]
Looks good, just needs an update to resources.yaml. You can run the tests for the apm-data plugin with these commands:

```
./gradlew x-pack:plugin:apm-data:test
./gradlew x-pack:plugin:apm-data:yamlRestTest
```

File: x-pack/plugin/apm-data/src/main/resources/component-templates/logs-apm@settings.yaml
Comment (endorama) [reply x-pack/plugin/apm-data/src/main/resources/component-templates/logs-apm@settings.yaml:1]: Thank you, done!

Comment (endorama) [reply x-pack/plugin/apm-data/src/main/resources/component-templates/logs-apm@settings.yaml:1]: I also added a commit documenting this process in the README for future reference.

File: docs/changelog/108862.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+pr: 108862
+summary: "Apm-data: set codec"
```
Comment (axw) [inline docs/changelog/108862.yaml:2]:
```suggestion
summary: "Apm-data: set codec: best_compression for logs-apm.* data streams"
```

File: x-pack/plugin/apm-data/src/main/resources/component-templates/logs-apm@settings.yaml
Comment (axw) [reply x-pack/plugin/apm-data/src/main/resources/component-templates/logs-apm@settings.yaml:1]: Thanks, good call.

**Review by axw — APPROVED** [review]

**Pull request merged**